### ISSUE TITLE
Modify G65 to support implementing current location in a breadcrumb trail as a link with `aria-current="page"`

### ISSUE THREAD

**Issue by mcking65** [w3c/wcag#3948]
The tests in [G65: Providing a breadcrumb trail | WAI | W3C](https://www.w3.org/WAI/WCAG22/Techniques/general/G65)
prohibit authors from implementing the current location in a breadcrumb trail as a link.

>5. For a breadcrumb trail that does include the current location:
>    1. Check that all elements except for the current location are implemented as links.
>     2. Check that the current location is not implemented as a link.

This is a proposal to change the G65 tests so the current location can optionally be represented by a link with `aria-current="page"` by revising the check as follows:

>5. For a breadcrumb trail that does include the current location:
>    1. Check that all elements prior to the current location are implemented as links.
>     2. Check that the current location is either not implemented as a link or is implemented as a link with `aria-current="page"`.

The revised technique could also reference the ARIA Authoring Practices Guide for a [supporting example](https://www.w3.org/WAI/ARIA/apg/patterns/breadcrumb/examples/breadcrumb/).

#### Why?

1. There are valid accessibility reasons for implementing the current location as a link.
2. WCAG techniques are more valuable and robust when they avoid unnecessarily restricting design and facilitate maximum authoring flexibility.
3. It would improve harmony among W3C WAI resources. G65, as explained below, makes the breadcrumb trail example in the ARIA Authoring Practices Guide appear invalid.
 
#### Rationale

The [breadcrumb example in the ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/breadcrumb/examples/breadcrumb/)
implements the current location as a link.

The APG task force recently became aware of this difference between the APG and WCAG technique when the task force received this issue --
[Should the current location on a breadcrumb trail be an anchor element? · Issue #3047 · w3c/aria-practices](https://github.com/w3c/aria-practices/issues/3047)

The APG task force chose to use a link for current location because:
1. The markup clearly communicates the current location in the navigation using `aria-current`.
2. The design makes current location perceivable when tabbing or reading a list of links.

If the current location is plain text, the meaning of the text depends entirely on the surrounding context. While that is not an unreasonable dependency, it is an unnecessary dependency for the breadcrumb pattern because the dependency can be removed by using a link with `aria-current` instead of plain text.

**Comment by patrickhlauke** [comment]
The thing with techniques is though: not following the technique does not mean an authors necessarily fails the SC. Techniques are but one way of passing an SC. They're not comprehensive or complete, so the current G65 does not preclude authors from actually using `aria-current` and having the current page as a link.

**Comment by scottaohara** [comment]
seems G65 could just be modified a bit to allude to the fact that either are ok.

**Comment by fstrr** [comment]
That reminds me that [I wrote a technique for `aria-current` in 2021](https://github.com/w3c/wcag/pull/1707).
